File: algoexpert/easy/09.product_sum.js
const input = [1, 2, 3, [1, 3], 3];

function product_sum(lst, depth) {
  let sum = 0;

  for(const el of lst) {
    if (Array.isArray(el)) {
      sum += product_sum(el, depth+1)
    } else {
      sum += el;
    }
  }

  return sum * depth;
}

console.log("sum product is: " + product_sum(input, 1));

File: algoexpert/easy/09.product_sum.lua
function product_sum(lst, sum, depth)
   for _, val in pairs(lst) do
      if type(val) == "table" then
         sum = sum + product_sum(val, 0, depth+1)
      else
         sum = sum + val
      end
   end
   return sum * depth
end

print(product_sum({1, 2, 3, {1, 3}, 3}, 0, 1))
